gen3/bin/healthcheck.sh

```diff
@@ -0,0 +1,139 @@
+        terminatingPods=$(echo $terminatingPods | jq -r ". += [$pod]")
+      fi
+    fi
+  done <<< "$(echo $allPods | jq -c '.[]')"
+
+  # check status of nodes
+  # https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#nodestatus-v1-core
+  local allNodes=$(g3kubectl get nodes -o json | \
+    jq -r '[
+      .items[] | {
+        name: .metadata.labels."kubernetes.io/hostname", conditions: [ .status.conditions[] | { (.type): .status }] | add
+      }
+    ]')
+  local notReadyNodes=$(echo $allNodes | jq -r '[.[] | select(.conditions.Ready != "True")]')
+
+  # check internet access
+  local curlCmd="curl -s -o /dev/null -I -w "%{http_code}" https://www.google.com"
+  if [[ $HOSTNAME == *"admin"* ]]; then # if in admin vm, run curl in fence pod
+    curlCmd="g3kubectl exec $(get_pod fence) -- $curlCmd"
+  fi
+  local statusCode=$(eval $curlCmd)
+  local internetAccess=true
+  if [[ $statusCode -lt 200 || $statusCode -ge 400 ]]; then
+    internetAccess=false
+  fi
+  (
+    export http_proxy="http://cloud-proxy.internal.io:3128"
+    export https_proxy=$http_proxy
+    local statusCodeExplicit=$(eval $curlCmd)
+    if [[ $statusCodeExplicit -lt 200 || $statusCodeExplicit -ge 400 ]]; then
+      exit 1
+    else
+      exit 0
+    fi
+  )
+  explicitProxyResult=$?
+  local internetAccessExplicitProxy=true
+  if [[ $explicitProxyResult != 0 ]]; then
+    internetAccessExplicitProxy=false
+  fi
+
+  local healthy=true
+  if [[ "$internetAccess" = false || "$internetAccessExplicitProxy" = false || "$pendingLongPods" != "[]" || "$terminatingPods" != "[]" || "$unknownPods" != "[]" || "$crashLoopPods" != "[]" || "$evictedPods" != "[]" || "$notReadyNodes" != "[]" ]]; then
+    healthy=false
+  fi
+
+  local healthJson=$(echo '{}' | jq -r "{
+    pendingTimeoutPods: $pendingLongPods,
+    terminatingTimeoutPods: $terminatingPods,
+    unknownPods: $unknownPods,
+    crashLoopBackOffPods: $crashLoopPods,
+    evictedPods: $evictedPods,
+    notReadyNodes: $notReadyNodes,
+    internetAccess: $internetAccess,
+    internetAccessExplicitProxy: $internetAccessExplicitProxy
+  }")
+  echo $healthJson | jq -r '.'
+
+  if [[ "$HEALTH_SEND_SLACK" = true && "$healthy" = false ]]; then
+    if [[ "${slackWebHook}" == 'None' || -z "${slackWebHook}" ]]; then
+      slackWebHook=$(g3kubectl get configmap global -o jsonpath={.data.slack_webhook})
+    fi
+    if [[ "${slackWebHook}" == 'None' || -z "${slackWebHook}" ]]; then
+      echo "WARNING: slackWebHook is None or doesn't exist; not sending results to Slack"
+    else
+      local tmpHostname=$(g3kubectl get configmap manifest-global -o jsonpath={.data.hostname})
+      local formattedAttachment="{\"title\": \"Status\", \"text\": \"$(echo $healthJson | sed s/\"/\\\\\"/g | sed s/,/,\\n/g)\", \"color\": \"#FF0000\"}"
+      curl -X POST --data-urlencode "payload={\"text\": \"Healthcheck FAILED for ${tmpHostname}\n\", \"attachments\": [$formattedAttachment]}" "${slackWebHook}"
```

Comment:
If curling to the outside to the outside to check for internet access, this one will most likely fail as well, so maybe a timeout flag would also be useful here